jetty-server/src/main/java/org/eclipse/jetty/server/ContentProducer.java

```diff
@@ -0,0 +1,87 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+public interface ContentProducer
+{
+
+    void recycle();
+
+    /**
+     * This call is always non-blocking.
+     * Doesn't change state.
+     * @return true if EOF was reached.
+     */
+    boolean consumeAll(Throwable x);
+
+    void checkMinDataRate();
+
+    /**
+     * This call is always non-blocking.
+     * Doesn't change state.
+     */
+    long getRawContentArrived();
+
+    /**
+     * This call is always non-blocking.
+     * Doesn't change state.
+     */
+    int available();
+
+    /**
+     * This call is always non-blocking.
+     * Doesn't change state.
+     * Doesn't query the HttpChannel.
+     */
+    boolean hasContent();
+
+    /**
+     * This call is always non-blocking.
+     * Doesn't change state.
+     * Doesn't query the HttpChannel.
+     */
+    boolean isError();
+
+    /**
+     * This call may block.
```

Comment:
Because of poor javadoc. What I meant is that some implementations may decide to make `nextContent()` a blocking call, some others may not. This is exactly what `BlockingContentProducer` and `AsyncContentProducer` do but I'd like the javadoc to avoid speaking about specific implementations.